Project/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs

```diff
@@ -107,8 +140,22 @@ public void MoveAgent(float[] act)
 
     public override void OnActionReceived(float[] vectorAction)
     {
-        // Existential penalty for strikers.
-        AddReward(-1f / 3000f);
+
+        if (position == Position.Goalie)
+        {
+            // Existential bonus for Goalies.
+            AddReward(1f / 3000f);
+        }
+        else if (position == Position.Striker)
+        {
+            // Existential penalty for Strikers
+            AddReward(-1f / 3000f);
+        }
+        else
+        {
+            // Existential penalty cumulant for Generic
+            timePenalty += -1f / 3000f;
```

Comment:
You only accumulate this time penalty for the "generic" position. Is that intended or were you trying to give it to the strikers ?

Reply:
That was intended. The strikers deserve it every timestep